knip.jsonc

```diff
@@ -31,7 +31,11 @@
                 "eslint-config-eslint",
 
                 // Optional peer dependency used for loading TypeScript configuration files
-                "jiti"
+                "jiti",
+                // Optional peer dependencies used inside types
+                "@eslint/core",
+                "@types/estree",
+                "@types/json-schema"
```

Comment:
I actually agree with knip that these should not be optional `peerDependencies`. Optional peers are not installed by default by `npm install`, they only instruct npm to raise an error if another package tries to install those dependencies in a version that is out of the specified range. I'm not sure this is even desirable in this case.

Reply:
These should be optional because only a subset of users who want types would be using them.

Reply:
I agree with @fasttime, I think `@eslint/core`, `@types/estree` and `@types/json-schema` should just be `dependencies`. It especially makes sense for `@types/estree` and `@types/json-schema` since they were [previously](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1abf76892b6e171ea60ad2180ac3cd40fd01d1f6/types/eslint/package.json#L19C5-L22C7) `dependencies` of `@types/eslint` which has now been merged with `eslint`.

Reply:
For users it would be easier if ESLint depended on those packages as regular dependencies, but if we are going to ask them to install the packages manually in the docs, then we don't need to add anything to our own `package.json`. We could also include those packages as optional peer dependencies with version `"*"` so that they don't cause version conflicts like we did for `jiti`.

Reply:
I'm running ESLint v9.11.0, and I'm having TS related issues because I need to manually install `@eslint/core`. I think it'll make more sense if the packages we are directly deriving types from to be `dependencies`. Since we can't really throw a runtime error to let them know to install them like we did with `jiti`.

Reply:
Peer dependencies are automatically installed with `npm install`. Is there a reason we'd prefer dependencies over that?

Reply:
@nzakas Peer dependencies can clash between different versions and are harder to get updated through eg. Renovate and Dependabot – only reason to add these as peer dependencies is if they are considered too heavy to ship with ESLint by standard, making it an optional opt in for those who want to use types

Reply:
We can make these direct dependencies the only downside is that it would increase some bundle size, but I think it should be fine because in most cases ESlint would be installed as a dev dependency only. Similarly, packages like [webpack](https://github.com/webpack/webpack/blob/dae16ad11e2d2accddc549a92082da3d2e667b01/package.json#L8) and [webpack-dev-serve](https://github.com/webpack/webpack-dev-server/blob/6471cf851720cd5e3546618dc4179343d1fb62ae/package.json#L49-L52) have also mentioned `@types/*` directly in dependencies.